tensorly/metrics/factors.py

```diff
@@ -28,16 +33,33 @@ def congruence_coefficient(matrix1, matrix2, absolute_value=True):
     congruence : float
     permutation : list
     """
-    num_cols = T.shape(matrix1)[1]
-    if T.shape(matrix1) != T.shape(matrix2):
-        raise ValueError("Matrices must have same shape")
-
-    matrix1 = matrix1 / T.norm(matrix1, axis=0)
-    matrix2 = matrix2 / T.norm(matrix2, axis=0)
-    all_congruences = T.dot(T.transpose(matrix1), matrix2)
-    if absolute_value:
-        all_congruences = T.abs(all_congruences)
-    all_congruences = T.to_numpy(all_congruences)
+    # Check if matrix1 and matrix2 are lists of the same length
+    if isinstance(matrix1, list):
+        if not isinstance(matrix2, list) or len(matrix1) != len(matrix2):
+            raise ValueError("Input lists of matrices must have the same length")
+    else:
+        matrix1 = [matrix1]
+        matrix2 = [matrix2]
+    all_congruences_list = []
+    # Store an arbitrary column dimension, they should all be the same
+    num_col_mat1 = T.shape(matrix1[0])[1]
```

Comment:
We can also check all matrices have the same number of columns  
```python
columns = [T.shape(m) for m in matrix1] + [T.shape(m) for m in matrix2]
if len(np.unique(np.array(columns)) > 1:
    raise ValueError(...)
```

Reply:
Great snippet thanks ! I did check that by comparing all sizes to the first one, but your code is cleaner.